### ISSUE TITLE
Reviews for Two Scoops of Django: Best Practices for Django 1.8 by Daniel Roy Greenfeld

### ISSUE THREAD

**Issue by gutfeeling** [gutfeeling/pythonbooks_reviews#80]

**Comment by gutfeeling** [comment]
:star: :star: :star: :star: 

I read this book in 2014. At that time, I had already completed the [Official Django Tutorials](https://docs.djangoproject.com/en/1.11/intro/tutorial01/) and knew how to code a very simple app. However, I was not confident in my Django code and didn't know how to handle more complex projects. I figured out a few things from the [official documentation](https://docs.djangoproject.com/en/1.11/) but I was always worried that I was not following best practices. 

I picked up this book hoping that it will make me more confident with Django. I must say that the book was very effective in this regard. When I finished this book, I felt like I could advice others on how to organize a Django website. The authors cover important topics like how to use virtualenv, requirements file, different settings files for development and production, how to leverage model inheritance and class based views to stay DRY and much much more. It helps that the authors are recognized experts in Django, so that made me confident in the knowledge that I gained from the book.

### Things I liked about this book

- The book uses the example of an ice cream website throughout the book to illustrate the concepts. I really enjoyed the examples and could understand the concepts better in context.

- The writing style is beautiful. The book is a pleasure to read. 

### Things I didn't like

- I felt that the book went a little overboard with the DRY principle. I say this from experience as it encouraged me to make websites more complicated than they ought to be. However, this is not really a major downside as every book has its own angle and one needs to augment it with experience. None of the suggestions in the book are theoretically incorrect. 

In summary, if you have a little experience with Django and looking to solidify your knowledge, then this book is great. I recommend it wholeheartedly.
